src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -1265,7 +1265,9 @@ private static void RunResultsCachingTestCase(ResultsCachingTestCase testCase, b
                 OutputFilePath = testCase.PersistLogFileToDisk ? Guid.NewGuid().ToString() : null,
                 TargetFileSpecifiers = new string[] { Guid.NewGuid().ToString() },
                 Kind = new List<ResultKind> { ResultKind.Fail },
-                Level = new List<FailureLevel> { FailureLevel.Warning, FailureLevel.Error }
+                Level = new List<FailureLevel> { FailureLevel.Warning, FailureLevel.Error },
+                AutomationId = "Test Automation",
+                AutomationGuid = Guid.NewGuid().ToString()
```

Comment:
>AutomationGuid [](http://example.com/codeflow?start=16&length=14)

So you don't have explicit tests somewhere for passing only one of these, or none.

Also, you've dropped these tests into an existing test, which is not necessarily bad, but again, we don't have a directed test. 

Did you cover the multithreaded case? Please verify with code coverage that you have coverage of all your changes.

Reply:
just added a new test that will cover the following scenarios:
- id only
- guid only
- id and guid

Also, the test verifies single and multithread.